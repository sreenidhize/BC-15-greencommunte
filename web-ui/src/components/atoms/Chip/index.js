import React from "react";
import MUIChip from "@material-ui/core/Chip";
import CloseIcon from "@material-ui/icons/Close";
import { makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
  root: {
    color: theme.palette.primary.main,
    borderColor: theme.palette.primary.main,
    padding: theme.spacing(1, 1.2),
    fontFamily: theme.typography.fontFamily,
    height: "38px",
    borderRadius: "6px",
    "& .MuiChip-label": {
      maxWidth: 100,
    },
  },
  icon: {
    color: theme.palette.primary.main,
  },
}));

const Chip = (props) => {
  const classes = useStyles();
  const { content, handleDelete } = props;
  return (
    <div data-testid="chip-placeholder">
      {content && (
        <MUIChip
          variant="outlined"
          label={content}
          deleteIcon={<CloseIcon className={classes.icon} />}
          size="small"
          className={classes.root}
          onDelete={handleDelete}
        />
      )}
    </div>
  );
};

Chip.propTypes = {
  content: PropTypes.string.isRequired,
};

Chip.defaultProps = {
  content: "chip",
};

export default Chip;
