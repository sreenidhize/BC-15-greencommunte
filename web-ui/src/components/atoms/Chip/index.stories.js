import React from "react";
import Chip from ".";

export default {
  title: "atoms/Chip",
  component: Chip,
};

const Template = (args) => <Chip {...args} />;

const alertFunction = () => {
  alert("hello");
};

export const Primary = Template.bind({});
Primary.args = {
  // content: "hello",
  deleteFunction: alertFunction,
};
