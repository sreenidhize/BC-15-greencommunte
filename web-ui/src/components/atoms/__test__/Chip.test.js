import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Chip from "../Chip";

describe("Chip placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <Chip />
    );
    expect(container).toMatchSnapshot();
  });

  global.fetch = jest.fn(() =>
  Promise.resolve()
);

  test("test chip render", () => {
    const { getByTestId } = render(<Chip />);
    expect(getByTestId("chip-placeholder")).toBeInTheDocument();
  });
});