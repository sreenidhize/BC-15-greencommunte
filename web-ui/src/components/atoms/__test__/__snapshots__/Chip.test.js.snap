// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip placeholder test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="chip-placeholder"
  >
    <div
      class="MuiChip-root makeStyles-root-1 MuiChip-outlined MuiChip-sizeSmall"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        chip
      </span>
    </div>
  </div>
</div>
`;
